### PR TITLE
feat(storage): support autoclass v2.1 samples

### LINKED ISSUE #3314: storage: update autoclass samples for v2.1
Update Autoclass samples for (1) new fields introduced in v2.1 and (2) remove v1 comment. Full details are available in this [internal doc](https://docs.google.com/document/d/11MqjaM94_mn0pFSLF9LYdbcAQWGw3IKfUhazZrXJSu0/edit?resourcekey=0-Imw-PsXwtQ0e18C9a1drxw#heading=h.vhgf40fpx2jn).

Please wait for approval from myself before merging/releasing so we can [coordinate](http://goto.google.com/gcs-sdk-ac2-1) around release dates.

### PR DESCRIPTION
**Pull request by cojenco**
Fixes #3314

Requires https://github.com/googleapis/google-cloud-go/pull/8721

### CODE REVIEW

**Review by tritone — COMMENTED** [review]
One minor thing, otherwise LGTM once the library PR is merged & released.

File: storage/buckets/set_autoclass.go
Comment (cojenco) [reply storage/buckets/set_autoclass.go]: I realized I missed this last time, fixed minimal arguments

**Comment by cojenco** [comment]
@tritone ready for review and merge now that storage is at the latest release, thanks!

**Review by tritone — APPROVED** [review]

**Pull request merged**
